### PR TITLE
ODM-11357 sdk bearer token support

### PR DESCRIPTION
**Pull request by natalyatodorova**

### CODE REVIEW

File: odm_sdk/connection.py [hunk 1]
```diff
@@ -131,6 +131,24 @@ def login_by_token(self, token):
             hostname = urlsplit(self.server_url).hostname
             raise GenestackAuthenticationException('Fail to login by token to %s' % hostname)
 
+    def login_by_access_token(self, access_token):
+        """
+        Attempt a login on the connection with the specified access token.
+        Raises an exception if the login fails.
+
+        :param access_token: OAuth access token
```
Comment (m-smazhevsky) [inline odm_sdk/connection.py:139]: :type access_token: str

File: odm_sdk/connection.py [hunk 2]
```diff
@@ -131,6 +131,24 @@ def login_by_token(self, token):
...
+        """
+        access_token = os.getenv(access_token, access_token)
+        self.session.headers.update({'Authorization': f'Bearer {access_token}'})
+        logged = self.application('genestack/signin').invoke('authenticateOAuthAccessToken', access_token)
+        Connection._handle_inactive_user_auth(logged)
+        if not logged['authenticated']:
+            self.session.headers.pop('Authorization')
+            hostname = urlsplit(self.server_url).hostname
+            raise GenestackAuthenticationException('Fail to login by token to %s' % hostname)
+
```
Comment (m-smazhevsky) [inline odm_sdk/connection.py:152]: [Optional] extract duplicated code

File: odm_sdk/scripts/user_setup.py [hunk 3]
```diff
@@ -55,11 +55,14 @@ def create_user_from_input(host, alias):
...
-    use_token = interactive_select(items, 'Select authentication') == by_token
+    by_access_token = 'by access token'
+    items = [by_token, by_access_token, 'by email and password']
+    choice = interactive_select(items, 'Select authentication')
 
-    if use_token:
+    if choice == by_token:
         return create_user_from_token(host, alias=alias)
+    elif choice == by_access_token:
+        return create_user_from_access_token(host, alias)
```
Comment (m-smazhevsky) [inline odm_sdk/scripts/user_setup.py]:
```suggestion
        return create_user_from_access_token(host, alias=alias)
```

File: odm_sdk/utils.py [hunk 4]
```diff
@@ -69,6 +71,9 @@ def make_connection_parser(user=None, password=None, host=None, token=None):
                        help='user password')
     group.add_argument('--token', dest='token', default=token, metavar='<api-token>',
                        help='API token to be used instead of the login and password')
+    group.add_argument('--access-token', dest='access_token', default=access_token, metavar='<access-token>',
+                       help='Access token or environment variable with its value to be used instead '
```
Comment (m-smazhevsky) [inline odm_sdk/utils.py:75]: smth about oauth? Let's double check with product guys

Comment (natalyatodorova) [reply odm_sdk/utils.py:75]: asked Mikhail Afanasev

File: odm_sdk/connection.py (hunk 2 above)
Comment (natalyatodorova) [reply odm_sdk/connection.py:152]: I supposed we are going to remove interactive mode from user, so there wouldn't be duplicates

File: odm_sdk/utils.py (hunk 4 above)
Comment (MikhailAf) [reply odm_sdk/utils.py:75]: @natalyatodorova @m-smazhevsky it seems okay for me

File: odm_sdk/connection.py (hunk 2 above)
Comment (natalyatodorova) [reply odm_sdk/connection.py:152]: created https://genestack.atlassian.net/browse/ODM-11600 for this

File: odm_sdk/settings/user.py [hunk 5]
```diff
@@ -26,6 +26,29 @@ def _get_server_url(host):
         "".format(host))
 
 
+def _read_non_canonical(msg):
```
Comment (m-smazhevsky) [inline odm_sdk/settings/user.py:29]: please add a small description wtf is that :)

**Review by m-smazhevsky — DISMISSED** [review]

File: odm_sdk/settings/user.py (hunk 5 above)
Comment (natalyatodorova) [reply odm_sdk/settings/user.py:29]: 😬

Comment (natalyatodorova) [reply odm_sdk/settings/user.py:29]: done

**Review by m-smazhevsky — APPROVED** [review]

**Pull request merged**